client_test.go

```
@@ -20,26 +23,31 @@ func TestMain(m *testing.M) {
 	token = os.Getenv("ABLY_ACCOUNT_TOKEN")
 	rand.Seed(time.Now().UnixNano())
 
-	if token == "" {
-		panic("ABLY_ACCOUNT_TOKEN not set")
-	}
-
 	if os.Getenv("ABLY_CONTROL_URL") != "" {
 		url = os.Getenv("ABLY_CONTROL_URL")
 	}
 
-	// Attempt to clean up apps if anything went wrong
-	client, _, err := NewClientWithURL(token, url)
-	if err == nil {
-		for _, v := range apps {
-			_ = client.DeleteApp(v)
+	// Attempt to clean up apps if anything went wrong (only if token is set)
+	if token != "" {
+		client, _, err := NewClientWithURL(token, url)
+		if err == nil {
+			for _, v := range apps {
+				_ = client.DeleteApp(v)
+			}
 		}
 	}
 
 	code := m.Run()
 	os.Exit(code)
 }
 
+// skipIfNoToken skips a test if ABLY_ACCOUNT_TOKEN is not set.
+func skipIfNoToken(t *testing.T) {
+	if token == "" {
+		t.Skip("ABLY_ACCOUNT_TOKEN not set, skipping integration test")
```

Does this have the potential to silently skip all integration tests if there's a configuration error and `ABLY_ACCOUNT_TOKEN` becomes unset in CI?

---

It would skip them.
I personally consider it exceedingly unlikely that the token would become unset (and that would arguably be indicative of more significant problems in our config change process), but I can switch out the account token check to check for a different flag like "SKIP_INTEGRATION" if that would help?
It was added mainly as a way to make testing, when I didn't need to check functions that call the API, quicker

---

Yeh I think I'd prefer a flag which explicitly skips them that can be used locally.